lib/identity_cache/should_use_cache.rb

```diff
@@ -0,0 +1,13 @@
+module IdentityCache
+  module ShouldUseCache
+    extend ActiveSupport::Concern
+
+    included do
+      unless self.respond_to?(:should_use_cache?)
+        def self.should_use_cache?
```

Comment:
I'm concerned about this scenario:

```ruby
class AssociatedRecord < ActiveRecord::Base
  def self.should_use_cache?
    # some logic
  end
end

class Thing < ActiveRecord::Base
  include IdentityCache
  # this will override `AssociatedRecord#should_use_cache?`
  cache_has_many :associated_record 
end
```